server/src/domain/metadata/metadata.service.ts

```diff
@@ -340,6 +340,31 @@ export class MetadataService {
     const stats = await this.storageRepository.stat(asset.originalPath);
     const mediaTags = await this.repository.getExifTags(asset.originalPath);
     const sidecarTags = asset.sidecarPath ? await this.repository.getExifTags(asset.sidecarPath) : null;
+
+    const dateDateSidecarTags = firstDateTime(sidecarTags as Tags, [
+      'SubSecDateTimeOriginal',
+      'DateTimeOriginal',
+      'SubSecCreateDate',
+      'CreationDate',
+      'CreateDate',
+      'SubSecMediaCreateDate',
+      'MediaCreateDate',
+      'DateTimeCreated',
+    ]);
+
+    if (dateDateSidecarTags && mediaTags) {
+      // remove time from mediaTags, since sidecarTags has a more accurate time
+
+      mediaTags.SubSecDateTimeOriginal = undefined;
+      mediaTags.DateTimeOriginal = undefined;
+      mediaTags.SubSecCreateDate = undefined;
+      mediaTags.CreationDate = undefined;
+      mediaTags.CreateDate = undefined;
+      mediaTags.SubSecMediaCreateDate = undefined;
+      mediaTags.MediaCreateDate = undefined;
+      mediaTags.DateTimeCreated = undefined;
```

Comment:
Settings things to undefined when they are being merged might lead to weird results in the future.

```typescript
const a = { CreateDate: 'placeholder' };
const b = { CreateDate: undefined };
const c = { ...a, ...b };
```
In this example c.CreateDate will actually be undefined, which I feel like is confusing and misleading. What do you think about just deleting from mediaTags list itself? So, `delete mediaTags.CreateDate` instead?

Reply:
This is actually the same list of tags as above, so I think these need to be kept in sync. Maybe we can pass a list above, and then loop over this list here and delete the values.

Reply:
> Settings things to undefined when they are being merged might lead to weird results in the future.
> 
> ```ts
> const a = { CreateDate: 'placeholder' };
> const b = { CreateDate: undefined };
> const c = { ...a, ...b };
> ```
> 
> In this example c.CreateDate will actually be undefined, which I feel like is confusing and misleading. What do you think about just deleting from mediaTags list itself? So, `delete mediaTags.CreateDate` instead?

look like a better idea indeed !